### PR TITLE
[autoscaler] port-forward for attach + redis_port

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Adds "port-forward" option for attach. Also adds shortcut "-p".

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Comment by richardliaw** [comment]
cc @eugenevinitsky

**Comment by eugenevinitsky** [comment]
Thanks! I've pinged the people in Flow who asked about this.

File: python/ray/tests/test_ray_init.py [hunk 1]
```diff
@@ -58,6 +58,20 @@ def f():
...
 
+    def test_redis_port(self, shutdown_only):
+        @ray.remote
+        def f():
+            return 1
+
+        info = ray.init(redis_port=1234, redis_password="testpassword")
+        address = info["redis_address"]
+        redis_ip, redis_port = address.split(":")
+        assert redis_port == 1234
```
Comment (richardliaw) [inline python/ray/tests/test_ray_init.py]:
```suggestion
        assert redis_port == "1234"
```

**Review by edoakes — APPROVED** [review]
LGTM

**Pull request merged**